modules/integration/tests/src/test/java/org/wso2/appmanager/integration/test/cases/RatingTestCase.java

```diff
@@ -0,0 +1,128 @@
+*software distributed under the License is distributed on an
+*"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+*KIND, either express or implied.  See the License for the
+*specific language governing permissions and limitations
+*under the License.
+*/
+package org.wso2.appmanager.integration.test.cases;
+
+import org.json.JSONObject;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.appmanager.integration.utils.APPMPublisherRestClient;
+import org.wso2.appmanager.integration.utils.APPMStoreRestClient;
+import org.wso2.appmanager.integration.utils.AppmTestConstants;
+import org.wso2.appmanager.integration.utils.bean.SubscriptionRequest;
+import org.wso2.carbon.automation.engine.context.AutomationContext;
+import org.wso2.carbon.automation.engine.context.TestUserMode;
+import org.wso2.carbon.automation.engine.context.beans.User;
+import org.wso2.carbon.automation.test.utils.http.client.HttpResponse;
+
+public class RatingTestCase {
+    private static final String TEST_DESCRIPTION = "Test web application ratings";
+    private APPMPublisherRestClient appmPublisherRestClient;
+    private APPMStoreRestClient appmStoreRestClient;
+    private AutomationContext appMServer;
+    private String appName = "RatingTestCase";
+    private String appVersion = "1.0.0";
+    private String context = "/" + appName;
+    private String trackingCode = "AM_" + appName;
+    private User adminUser;
+    private String userName;
+    private String password;
+    private String backEndUrl;
+
+
+    @BeforeClass(alwaysRun = true)
+    public void startUp() throws Exception {
+        appMServer = new AutomationContext(AppmTestConstants.APP_MANAGER,
+                                           TestUserMode.SUPER_TENANT_ADMIN);
+        backEndUrl = appMServer.getContextUrls().getWebAppURLHttps();
+        appmPublisherRestClient = new APPMPublisherRestClient(backEndUrl);
+
+        adminUser = appMServer.getSuperTenant().getTenantAdmin();
+        userName = adminUser.getUserName();
+        password = adminUser.getPassword();
+
+        appmPublisherRestClient.login(userName, password);
+        appmStoreRestClient = new APPMStoreRestClient(backEndUrl);
+    }
+
+    @Test(description = TEST_DESCRIPTION)
+    public void testWebAppRating() throws Exception {
+        HttpResponse response = appmPublisherRestClient.webAppCreate(appName, context, appVersion,
+                                                                     trackingCode);
+        JSONObject responseData = new JSONObject(response.getData());
+        String uuid = responseData.getString(AppmTestConstants.ID);
+        appmPublisherRestClient.publishWebApp(uuid);
+
+        final int ratingValue1 = 3;
+
+        // Subscribe and rate to app from admin user
+        HttpResponse adminUserResponse = subscribeAndRateApplication(adminUser, appName, userName,
```

Comment:
Fixed in https://github.com/wso2/product-app-manager/commit/f021f86c3c398dddeeca17e225a37e42f7f7033b